pkg/controller/controller.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2017 by the contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	authenticatorV1alpha1 "github.com/kubernetes-sigs/aws-iam-authenticator/pkg/apis/authenticator.aws/v1alpha1"
+	authenticatorclient "github.com/kubernetes-sigs/aws-iam-authenticator/pkg/client/clientset/versioned"
+	authenticatorinformerV1alpha1 "github.com/kubernetes-sigs/aws-iam-authenticator/pkg/client/informers/externalversions/authenticator.aws/v1alpha1"
+	"github.com/kubernetes-sigs/aws-iam-authenticator/pkg/config"
+	"github.com/kubernetes-sigs/aws-iam-authenticator/pkg/mappings"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/workqueue"
+
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+
+// New creates a new controller to listen to CRDs
+func New(c config.Config, identityMapping *mappings.Mappings) (*Controller, error) {
+	_, clientset, authenticatorset, err := createClient(c.Kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	iamidentityInformer := authenticatorinformerV1alpha1.NewIAMIdentityMappingInformer(authenticatorset, "kube-system", 10*time.Minute, cache.Indexers{})
+	ctrl := &Controller{
+		config:              c,
+		clientset:           clientset,
+		authenticatorset:    authenticatorset,
+		iamIdentityInformer: iamidentityInformer,
+		queue:               workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "authenticatorsync"),
+	}
+
+	logrus.Print("setting up iamidentitymapping listeners")
+	iamidentityInformer.AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				identity := obj.(*authenticatorV1alpha1.IAMIdentityMapping).DeepCopy()
+				logrus.Infof("iamidentitymapping %s added", identity.Name)
+				identityMap := mappings.Identity{
```

Comment:
This should be called a mapping (or identity), i.e. a single entry, whereas the entire data structure should be called the map or identityMap.